data/scraper_analysis.Rmd

```diff
@@ -0,0 +1,328 @@
+---
+title: "Article Quality Assessment"
+output: html_document
+---
+
+```{r, settings, echo=FALSE}
+knitr::opts_chunk$set(cache=FALSE, warning=FALSE,
+    message=FALSE, fig.align='center', fig.width=10)
+```
+
+```{r, load_library}
+library(RSQLite)
+library(dplyr)
+library(ggplot2)
+library(stringr)
+
+checkDuplicate = function(rawArticle, dataSource){
+    sub = 
+        rawArticle %>%
+        subset(., source == dataSource)
+
+    duplicateCount =
+        sub %>%
+        group_by(title) %>%
+        summarise(count = n()) %>%
+        arrange(., desc(count))
+    
+    cat("Data source: ", dataSource, "\n")
+    cat("Total number of entry: ", NROW(sub), "\n")
+    cat("Unique Titles: ", length(unique(sub$title)), "\n")
+    cat("Most duplicated title is: '", unlist(duplicateCount[1, "title"]), "'\n", sep = "")
+    cat("Duplicate count summary:\n")
+    print(summary(duplicateCount$count))
+    cat("\n")
+}
+
+getDuplicated = function(rawArticle, dataSource, n = 1){
+
+    sub = 
+        rawArticle %>%
+        subset(., source == dataSource)
+
+    duplicateCount =
+        sub %>%
+        group_by(title) %>%
+        summarise(count = n()) %>%
+        arrange(., desc(count))
+
+    
+    sub %>%
+        subset(., title == unlist(duplicateCount[n, "title"]))
+}
+
+```
+
+
+```{r, load_data}
+dataDir = Sys.getenv("DATA_DIR")
+dbName = "/the_reading_machine.db"
+fullDbPath = paste0(dataDir, dbName)
+con = dbConnect(drv=SQLite(), dbname=fullDbPath)
+statement = paste0('SELECT * FROM RawArticle')
+rawArticles =
+    dbGetQuery(con, statement) %>%
+    mutate(date = as.Date(date, format="%Y-%m-%d"))
+```
+
+There are a few observations we can make in the following graph. First
+of all, we can see that the Bloomberg data does not start until the
+end of 2011. Further, the number of articles remain low prior to early
```

Comment:
ok, at this stage, I will examine the model based solely on `Agrimoney` and `Euractiv`. Then I will assess the influence of the Bloomberg articles to see whether we will need the additional data.